tools/dep_updaters/update-openssl.sh

```diff
@@ -9,45 +9,115 @@ cleanup() {
   exit $EXIT_CODE
 }
 
-download() {
-  if [ -z "$1" ]; then
-    echo "Error: please provide an OpenSSL version to update to"
-    echo "	e.g. ./$0 download 3.0.7+quic1"
-    exit 1
-  fi
+download_v1() {
+  LATEST_V1_TAG_NAME="$("$NODE" --input-type=module <<'EOF'
+const res = await fetch('https://api.github.com/repos/quictls/openssl/git/matching-refs/tags/OpenSSL_1');
```

Comment:
Prior to OpenSSL 3, OpenSSL did not follow semver, so we can exclude 1.0.x and 1.1.0 and only look at the 1.1.1 tags.
```suggestion
const res = await fetch('https://api.github.com/repos/quictls/openssl/git/matching-refs/tags/OpenSSL_1_1_1');
```